### PR TITLE
NTSL update

### PR DESCRIPTION
**Pull request by Markolie**
Changes:
1) Fixes the $source variable not changing. Fixes https://github.com/ParadiseSS13/Paradise/issues/1021.
2) Adds implode(list, string), min(...), max(...), randseed(number), rand(), proper(string), time() and timestamp() functions. Ported from https://github.com/tgstation/-tg-station/pull/4927.
3) Fixes the broken / NTSL division operator, and adds sin(), cos(), asin(), acos() and log() functions. Ported from https://github.com/tgstation/-tg-station/pull/6786.
4) Adds a $language variable to NTSL (can not be modified), which returns the spoken language.
5) Minor reshuffle of the telecomms power room so the glass/cables no longer fly north when the game starts.

### CODE REVIEW

File: code/__HELPERS/time.dm [hunk 1]
```diff
@@ -4,25 +4,14 @@
...
 	return time2text(world.timeofday, "hh:mm:ss")
-
-/* Preserving this so future generations can see how fucking retarded some people are
-proc/time_stamp()
-	var/hh = text2num(time2text(world.timeofday, "hh")) // Set the hour
-	var/mm = text2num(time2text(world.timeofday, "mm")) // Set the minute
-	var/ss = text2num(time2text(world.timeofday, "ss")) // Set the second
-	var/ph
-	var/pm
-	var/ps
```
Comment (tigercat2000) [inline code/__HELPERS/time.dm]: Why did you remove this? The comment specifically stated it was there as an example of terribad code.

Comment (Markolie) [inline code/__HELPERS/time.dm]: Yes, I'm sure this'll educate millions on how to properly code.

File: code/modules/mob/hear_say.dm [hunk 2]
```diff
@@ -115,12 +115,6 @@
 
 	if(vname)
 		speaker_name = vname
-
-	if(istype(speaker, /mob/living/carbon/human))
-		var/mob/living/carbon/human/H = speaker
-		if(H.voice)
-			speaker_name = H.voice
```
Comment (tigercat2000) [inline code/modules/mob/hear_say.dm]: What even is the voice var used for?

File: code/__HELPERS/time.dm (hunk 1 above)
Comment (tigercat2000) [inline code/__HELPERS/time.dm]: @Markolie Mark plis, it's great. Don't remove it for the sake of.. like 5kb.

File: code/modules/mob/hear_say.dm (hunk 2 above)
Comment (Markolie) [inline code/modules/mob/hear_say.dm]: That check's already made in the radio's signal code. Basically, voice is updating on every life() and determines the voice of the character. By default real_name, unless masked, then name, unless they have a voicechanger/mimic, then the name of that.

**Comment by Spacemanspark** [comment]
:+1: Maybe the Tcomms mute script I use will work better now.

**Comment by Fox-McCloud** [comment]
Huzzah :+1: 

Will be good to get telecomms back on its feet.

**Comment by fludd12** [comment]
Oh, there's a variable tied to most telecomm signals that's not accessible via coding, which returns the race of the speaker. This could make VoxComms come back with double strength! (/s). If there's any way you could shoehorn this in eventually, I'd be glad.

**Comment by Robertha89** [comment]
yes plz, so we can be RACIST against Kidan and catbeast.
